tests/BasicTest.php

```diff
@@ -4,65 +4,81 @@
 
 namespace Atk4\Validate\Tests;
 
-use Atk4\Core\AtkPhpunit;
 use Atk4\Data\Model;
-use Atk4\Data\Persistence;
+use Atk4\Data\Schema\TestCase;
 use Atk4\Validate\Validator;
 
-class BasicTest extends AtkPhpunit\TestCase
+class BasicTest extends TestCase
 {
-    /** @var Model */
-    public $m;
-
-    /** @var Validator */
-    public $c;
-
     protected function setUp(): void
     {
-        $this->m = $m = new Model(new Persistence\Array_());
+        parent::setUp();
+
+        $this->setDb([
+            'validator_test' => [
+                1 => [
+                    'name' => 'John',
+                    'age' => 22,
+                    'type' => 'dog',
+                    'tail_length' => 5,
+                ],
+            ],
+        ]);
+    }
 
-        $m->addField('name');
-        $m->addField('age', ['type' => 'number']);
-        $m->addField('type', ['required' => true, 'enum' => ['dog', 'ball']]);
-        $m->addField('tail_length', ['type' => 'number']);
+    protected function getModel(): Model
+    {
+        return new ModelTest($this->db);
+    }
 
-        $this->c = new Validator($m);
+    protected function getValidator(Model $model): Validator
```

Comment:
also here, `createValidator`, programmer should be aware from the naming instantly what it does/returns